src/core/src/op/scatter_nd_update.cpp

```diff
@@ -140,5 +144,120 @@ bool ScatterNDUpdate::evaluate_symbol(TensorSymbolVector& output_symbols) const
     return default_symbol_evaluator(this, {0, 2}, output_symbols);
 }
 }  // namespace v3
+
+namespace v14 {
+ScatterNDUpdate::ScatterNDUpdate(const Output<Node>& inputs,
+                                 const Output<Node>& indices,
+                                 const Output<Node>& updates,
+                                 const ScatterNDUpdate::Reduction reduction)
+    : op::util::ScatterNDBase(inputs, indices, updates),
+      m_reduction{reduction} {
+    constructor_validate_and_infer_types();
+}
+std::shared_ptr<Node> ScatterNDUpdate::clone_with_new_inputs(const OutputVector& new_args) const {
+    OV_OP_SCOPE(v14_ScatterNDUpdate_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return std::make_shared<ScatterNDUpdate>(new_args.at(0), new_args.at(1), new_args.at(2), m_reduction);
+}
+
+bool ScatterNDUpdate::visit_attributes(AttributeVisitor& visitor) {
+    OV_OP_SCOPE(v14_ScatterNDUpdate_visit_attributes);
+    visitor.on_attribute("reduction", m_reduction);
+    return true;
+}
+
+bool ScatterNDUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+    OV_OP_SCOPE(v14_ScatterNDUpdate_evaluate);
+    OPENVINO_ASSERT(inputs.size() == 3);
+    OPENVINO_ASSERT(outputs.size() == 1);
+
+    const auto& data = inputs[0];
+    const auto& indices = inputs[1];
+    const auto& updates = inputs[2];
+    auto& output = outputs[0];
+    const auto& data_shape = data.get_shape();
+    const auto& indices_shape = indices.get_shape();
+    const auto& updates_shape = updates.get_shape();
+    output.set_shape(data_shape);
+    using namespace ov::element;
+    return IF_TYPE_OF_CONVERT_TENSORS(v14_ScatterNDUpdate_evaluate,
+                                      this,
+                                      outputs,
+                                      inputs,
+                                      OV_PP_ET_LIST(boolean, f32, i32, i64, u32, u64),
+                                      scatter_nd_update::Evaluate,
+                                      data.get_element_type(),
+                                      data,
+                                      indices,
+                                      updates,
+                                      output,
+                                      data_shape,
+                                      indices_shape,
+                                      updates_shape,
+                                      m_reduction);
+}
+
+bool ScatterNDUpdate::has_evaluate() const {
+    OV_OP_SCOPE(v14_ScatterNDUpdate_has_evaluate);
+
+    switch (get_output_element_type(0)) {
+    case element::boolean:
+    case element::f16:
+    case element::f32:
+    case element::i32:
+    case element::i64:
+    case element::u32:
+    case element::u64:
+        break;
+    default:
+        return false;
+    }
+    switch (get_input_element_type(1)) {
+    case element::i32:
+    case element::i64:
+        return true;
+    default:
+        return false;
+    }
```

Comment:
Looks same as for older version make local helper and share code (single helper or maybe separate for both each input).